### PR TITLE
Feature/implement form components

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
## Introduced storybook

I propose we use storybook for developing our 'dumb'-components (For instance an 'address-tile'). (See [this article](https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43)).
I think storybook is a great way to develop and document at the same time.

Installed the following add ons:
- Knobs ([docs](https://github.com/storybookjs/storybook/tree/master/addons/knobs))
   - Play around with component properties in storybook.
- Viewport ([docs](https://github.com/storybookjs/storybook/tree/master/addons/viewport))
   - Render storybook components in a set of preconfigured viewports.
   - I configured the same viewports as defined in amsterdam-asc.
- Docs ([docs](https://github.com/storybookjs/storybook/tree/master/addons/docs))
   - Document components.

## Introduced generic responsive props

Introduced a method of implementing generic responsive properties.

### Example:
```tsx
// Responsive properties.
// bgColor is blue on mobile, etc
<Foo bgColor={{ mobile: 'blue', tablet: 'red', desktop: 'green' }} />
// But you can also still do this:
// bgColor is green for all
<Foo bgColor='green' />
```

### How to implement:

``` tsx
const Foo = styled.div`
   ${(props: Props) => responsiveProps(props, {
      'bgColor': color => `background-color: ${color};`
      // ... you can define other props here
   })
`
```

## Introduced Box component

Use this component as an atomic building block.
You can set: `padding`, `margin`, `background-color`, `horizontal-align`, `vertical-align`, `width`. 
_All properties are responsive._

We can replace most separate defined styled components in our codebase with this component.

### Example
```tsx
<Box p={1} bgColor='tint1' width={{ mobileS: 12, tabletS: 6 }}>Foo</Box>
```

## Introduced amsterdam-react-final-form

Introduced `amsterdam-react-final-form`-components in the Generate-component.
`GenerateItineraryFormValues` is strongly typed, and directly passed into the `create` function.

### CODE REVIEW

File: src/components/atoms/Box/Box.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * Use box as your main building block.
+ * It can set padding, margin, width, backgroundColor, horizontal-align, and vertical-align.
+ * It can set these properties responsive as well.
+ *
+ * Example usage:
+ *
+ * ```
+ *    <Box padding={{ mobileS: 0, tabletM: 1 }} bgColor='level3' >
```
Comment (gilleswittenberg) [inline src/components/atoms/Box/Box.tsx:34]: Should `padding` not be `p`?

Comment (Jeffrey-Zutt) [reply src/components/atoms/Box/Box.tsx:34]: Yes :-)

File: src/components/atoms/Box/box.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const arrayToObject = (array: Array<string|number>) => array.reduce(
+  (acc, val) => ({ ...acc, [val]: val }),
+  { "undefined": undefined }
+)
+
+const oneToTwelve = arrayToObject([1,2,3,4,5,6,7,8,9,10,11,12])
+
+
+export const Example = () => (
+  // @ts-ignore
```
Comment (gilleswittenberg) [inline src/components/atoms/Box/box.stories.tsx:20]: Is this necessary?

Comment (Jeffrey-Zutt) [reply src/components/atoms/Box/box.stories.tsx:20]: Nope. Thanks!

**Pull request merged**